src/Message/RedirectPurchaseRequest.php

```diff
@@ -80,6 +85,7 @@ public function getData() : array
                 $card->getShippingCountry(),
                 'numeric'
             ),
+            static::HPP_ADDRESS_MATCH_INDICATOR => false,
```

Comment:
This is hard-coded to false (which is fine). I assume you didn't try and work out whether the two addresses were actually the same because it's too difficult (and it's hard to future-proof)?